### ISSUE TITLE
Error: Type Slide does not have resolver with name 'referenceMany'

### ISSUE THREAD

**Issue by davehinton-mg** [gridsome/gridsome#1180]
## Description

I'm getting an `Error: Type Slide does not have resolver with name 'referenceMany'` error, and it's not at all clear to me how to fix it.

### Steps to reproduce

1. Create a new Gridsome project by running `gridsome create project`.

2. Replace the `Gridsome.server.js` file with

        module.exports = function (api) {
          api.loadSource(async actions => {
            actions.addSchemaTypes(
              `
              type Course implements Node {
                id: ID!
                slides: [Slide!]!
              }
              type Slide implements Node {
                id: ID!
              }
              `
            );
          })
        }

3. In the `project` directory, run `gridsome develop`

The expected and actual results are the same if I swap the order in which I define `Slide` and `Course`, and if I pass an array of two strings (each containing a single type definition) instead of a single string.

This is a minimal example. In my real code, I have more fields on these types, and more types, and I load some JSON from disk and put it in a collection. But I get the same results as in the minimal example above.

### Expected result

No errors. Or, if there's something wrong with my schema types (I am new to GraphQL and to Gridsome, so this is more than possible), then an error message that does more to help me fix the thing that's wrong about the schema types I'm passing in.

### Actual result

I get this output:

    Gridsome v0.7.14

    Initializing plugins...
    Load sources - 0s
    Error: Type Slide does not have resolver with name 'referenceMany'
        at ObjectTypeComposer.getResolver (/Users/davidhinton/mg/publish-from-cosmic/node_modules/graphql-compose/lib/ObjectTypeComposer.js:903:13)
        at processObjectTypeFields (/Users/davidhinton/mg/publish-from-cosmic/node_modules/gridsome/lib/graphql/createSchema.js:319:42)
        at processTypes (/Users/davidhinton/mg/publish-from-cosmic/node_modules/gridsome/lib/graphql/createSchema.js:276:9)
        at createSchema (/Users/davidhinton/mg/publish-from-cosmic/node_modules/gridsome/lib/graphql/createSchema.js:64:3)
        at Schema.buildSchema (/Users/davidhinton/mg/publish-from-cosmic/node_modules/gridsome/lib/app/Schema.js:28:28)
        at Plugins.createSchema (/Users/davidhinton/mg/publish-from-cosmic/node_modules/gridsome/lib/app/Plugins.js:94:22)
        at processTicksAndRejections (internal/process/task_queues.js:97:5)

### Environment

      System:
        OS: macOS 10.15.4
        CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
      Binaries:
        Node: 14.2.0 - ~/.nvm/versions/node/v14.2.0/bin/node
        npm: 6.14.4 - ~/.nvm/versions/node/v14.2.0/bin/npm
      Browsers:
        Chrome: 81.0.4044.138
        Safari: 13.1
      npmPackages:
        gridsome: ^0.7.0 => 0.7.14
      npmGlobalPackages:
        @gridsome/cli: 0.3.2

**Comment by hjvedvik** [comment]
@davehinton-mg I tried your minimal example, but I'm only getting the same error message if I don't create the `Slide` collection. How do you create the collections in your project?

The error message should have been more informative though.

**Comment by davehinton-mg** [comment]
This was down to my misunderstanding, sorry.

In my real code, I was loading a single large blob of JSON as a `Course` collection, with this sort of structure:

```
[
  {
    someCourseAttribute: !String,
    slides: [
      {
        someSlideAttribute: !String,
        // etc
      }
    ]
  }
]
```

And this was fine initially, but some of the deeper type structure wasn't being inferred correctly, hence I felt I needed to provide an explicit schema.

I saw the warning in the docs that all my types had to implement `Node`, so I did that and added `id` attributes.

But I didn't appreciate that this meant I couldn't define the slides inline in the courses, that instead I had to hoist them out into a separate slides collection and in the courses reference them by id.

I had interpreted the error as meaning that there was something wrong with my schema, and did not realise that it was because there was no slides collection. (And in fairness, when I removed the courses collection (as part of constructing a minimal example), I didn't start to get an error about there being no courses resolver. Although I can see that this might be because courses aren't referenced in the schema.)

So in short: I had a faulty mental model, and the error message could have been more helpful.
